addons/mrp/models/mrp_bom.py

```diff
@@ -264,10 +248,9 @@ def _get_default_product_uom_id(self):
         digits='Product Unit of Measure', required=True)
     product_uom_id = fields.Many2one(
         'uom.uom', 'Product Unit of Measure',
-        default=_get_default_product_uom_id,
         required=True,
         help="Unit of Measure (Unit of Measure) is the unit of measurement for the inventory control", domain="[('category_id', '=', product_uom_category_id)]")
-    product_uom_category_id = fields.Many2one(related='product_id.uom_id.category_id')
+    product_uom_category_id = fields.Many2one(related='product_tmpl_id.uom_id.category_id')
```

Comment:
This is not needed as `product_tmpl_id` is already related to  `product_id.product_tmpl_id`
and also adding `product_tmpl_id` here will be slow as well.